2021.04+imx 5.10.35 2.0.0 fio: add imx6ulevk

MrCry0 opened the pull request:
add imx6ul-evk support

mike-scott committed:
[FIO extras] imx: mx6ul_14x14_evk: make USB configs specific to CONFIG_USB

Needed for having USB enabled even without CONFIG_USB_CMD (e.g.
fastboot).

Signed-off-by: Michael Scott <mike@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

mike-scott committed:
[FIO extras] imx: mx6ul_14x14_evk: allow setting CONFIG_BOOTCOMMAND in defconfig

Signed-off-by: Michael Scott <mike@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

mike-scott committed:
[FIO extras] imx: mx6ul_14x14_evk: dont include default env in SPL

This saves some space in the end binary.

Signed-off-by: Michael Scott <mike@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

mike-scott committed:
[FIO extras] board: imx6ul_14x14_evk: OP-TEE lower ram_size by OPTEE_TZDRAM_SIZE

Signed-off-by: Michael Scott <mike@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

mike-scott committed:
[FIO extras] imx: mx6ul_14x14_evk: add board_fit_config_name_match for SPL

When CONFIG_SPL_LOAD_FIT is enabled, the board must support the following
function:
int board_fit_config_name_match(const char *name)

Signed-off-by: Michael Scott <mike@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

mike-scott committed:
[FIO extras] dts: imx6ul-14x14-evk-u-boot: drop aliases in SPL dts

This saves over 2k in DTB size for SPL, and the aliases aren't needed.

Signed-off-by: Michael Scott <mike@foundries.io>
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

MrCry0 committed:
[FIO extras-squash] mx6ullevk: drop undefining MODULE_FUSE

In u-boot 2021.04 the selecting of IMX_MODULE_FUSE and OF_SYSTEM_SETUP
was disabled on Kconfig level. Drop obsolete undefining.

Fixes: 37f8fc5c10 ("[FIO extras] mx6ullevk: drop MODULE_FUSE and OF_SYSTEM_SETUP for SPL")
Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

MrCry0 committed:
[FIO internal] mx6: disable fuses and of-system setup in SPL

IMX_MODULE_FUSE adds ~8K of binary size to SPL where it's only used for 1
check to see if USB has been fused off.

For FIO use-case we only enable USB for mfgtool, where USB fusing would
not apply.

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

ricardosalveti commented on arch/arm/mach-imx/mx6/Kconfig:
```diff
@@ -61,7 +61,7 @@ config MX6SLL
 config MX6UL
 	bool "i.MX 6UltraLite SoC support"
-	select HAS_CAAM
+	select HAS_CAAM if !OPTEE
```
In theory this shouldn't really cause any issue as long the jr device tree node is marked as reserved by the secure world (done by optee), so we should really investigate why here (good to merge as is, but please create a jira so we can track it later).

MrCry0 replied:
Thanks, this fix shouldn't have been there. Fixed.

ricardosalveti reviewed (approved):
LGTM, minor comment but we can follow up later as the same change was already merged in 2020.04.

MrCry0 committed:
[FIO internal] misc: include misc-uclass depending on Kconfig option

There the SPL_MISC Kconfig option exists. Include misc-uclass driver
only if an option enabled.

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

MrCry0 committed:
Revert "[FIO internal-squash] cmd: fiovb: add include for boot_mode_is_closed"

This reverts commit aea6917404ce3bfd1b3c22110bd9ded9f00a82ad.

The commit duplicates another one which does the same:
f84c8adf48 ("[FIO extras-squash] cmd: fiovb: fix using undeclared function boot_mode_is_closed()")

Signed-off-by: Oleksandr Suvorov <oleksandr.suvorov@foundries.io>

vanmaegima reviewed (approved):
LGTM
